src/eventDispatcher.js

```diff
@@ -0,0 +1,35 @@
+function EventDispatcher() {
+    this.eventDispatcherMap = {};
+};
+
+EventDispatcher.prototype.bind = function(eventName, handler) {
+    if (this.eventDispatcherMap[eventName]) {
+        this.eventDispatcherMap[eventName].push(handler);
+    } else {
+        this.eventDispatcherMap[eventName] = [handler];
+    }
+};
+
+EventDispatcher.prototype.trigger = function(eventName) {
+    if (this.eventDispatcherMap[eventName]) {
+        for (var i = 0; i < this.eventDispatcherMap[eventName].length; i++) {
+            this.eventDispatcherMap[eventName][i]();
+        };
+    };
+};
+
+EventDispatcher.prototype.unbind = function(eventName, handler) {
+     if (this.eventDispatcherMap[eventName]) {
+           for (var i = 0; i < this.eventDispatcherMap[eventName].length; i++) {
+                if(this.eventDispatcherMap[eventName][i].toString() === handler.toString()) {
```

Comment:
This check does not make sense, why are you converting functions to a string? If you are using anonymous functions that wont event work. For example:
```
var f1 = function() { console.log("Hello from A") }
var f2 = function() { console.log("Hello from B") }
ed.bind("hello", f1);
ed.bind("hello", f2)
ed.unbind("hello", f1);
ed.trigger("hello");
// nothing will be printed here
```

Reply:
Do it simpler.